TypeScript issues with Node 16

Hi again,
I'm having trouble using your library with Node 16 and TypeScript. I'm working in a large monorepo so upgrading to Node 18 is not currently an option.

Including `@types/node` `16.11.20` in my dependencies and trying to type-check a script that uses your library is throwing these errors:

```
$ tsc --noEmit
node_modules/@dlenroc/testrail/src/internal/request.ts:70:34 - error TS2339: Property 'reason' does not exist on type 'AbortSignal'.

70     return Promise.reject(signal.reason);
                                    ~~~~~~

node_modules/@dlenroc/testrail/src/internal/request.ts:86:22 - error TS2339: Property 'reason' does not exist on type 'AbortSignal'.

86       reject(signal?.reason);
                        ~~~~~~

node_modules/@dlenroc/testrail/src/TestRail.ts:504:39 - error TS2339: Property 'reason' does not exist on type 'AbortSignal'.

504       return Promise.reject(ctxSignal.reason || options.signal.reason);
                                          ~~~~~~

node_modules/@dlenroc/testrail/src/TestRail.ts:504:64 - error TS2339: Property 'reason' does not exist on type 'AbortSignal'.

504       return Promise.reject(ctxSignal.reason || options.signal.reason);
                                                                   ~~~~~~

node_modules/@dlenroc/testrail/src/TestRail.ts:510:70 - error TS2339: Property 'reason' does not exist on type 'AbortSignal'.

510     const onAbort = () => { clearAbort(); controller.abort(ctxSignal.reason || optionsSignal.reason); };
                                                                         ~~~~~~

node_modules/@dlenroc/testrail/src/TestRail.ts:510:94 - error TS2339: Property 'reason' does not exist on type 'AbortSignal'.

510     const onAbort = () => { clearAbort(); controller.abort(ctxSignal.reason || optionsSignal.reason); };
                                                                                                 ~~~~~~


Found 6 errors in 2 files.

Errors  Files
     2  node_modules/@dlenroc/testrail/src/internal/request.ts:70
     4  node_modules/@dlenroc/testrail/src/TestRail.ts:504
error Command failed with exit code 2.
```

This seems to be because the `AbortSignal` interface changed between Node 16 and 18. I managed to pass your library the signal by hand (as opposed to using `AbortSignal.timeout()` as discussed in #70 since this isn't available in Node 16), so all my code passes type checks, but TypeScript seems to be checking the `.ts` files in your library despite my ignoring `node_modules` in `tsconfig.json`.

I created a minimum reproducible example here: https://github.com/treyturner/testrail-debug. You can see the problem I'm having by running `yarn type-check`. Is this a misconfiguration on my part?

I'm pretty inexperienced when it comes to JS/TS build configuration and package bundling; I see you ship both a `dist` folder and a `types.d.ts` file in your package. Are the `.ts` files needed? It occurs to me that if they weren't included, TypeScript wouldn't take issue with the package.

Thanks again for your time and work on the library.

--- a/package.json
+++ b/package.json
@@ -23,7 +23,7 @@
   "unpkg": "dist/TestRail.umd.js",
   "scripts": {
     "prepare": "npm run build",
-    "build": "rollup --bundleConfigAsCjs -c",
+    "build": "rollup --bundleConfigAsCjs -c && replace-in-file '/(\\.\\.\\/)+src/g' '../src' 'dist/*.map' --isRegex",
     "test": "mocha -A -r ts-node/register test/**/*.ts"
   },
   "engines": {
@@ -49,6 +49,7 @@
     "intermock": "^0.2.5",
     "mocha": "^10.2.0",
     "nock": "^13.3.1",
+    "replace-in-file": "^6.3.5",
     "rollup": "^3.21.6",
     "rollup-plugin-dts": "^5.3.0",
     "ts-node": "^10.9.1",

--- a/package.json
+++ b/package.json
@@ -15,7 +15,7 @@
     "src",
     "dist"
   ],
-  "types": "src/types.d.ts",
+  "types": "dist/TestRail.d.ts",
   "main": "dist/TestRail.node.js",
   "module": "dist/TestRail.node.mjs",
   "browser": "dist/TestRail.browser.mjs",
@@ -37,7 +37,7 @@
     "@faker-js/faker": "^7.6.0",
     "@rollup/plugin-terser": "^0.4.1",
     "@rollup/plugin-typescript": "^11.1.0",
-    "@types/chai": "^4.3.4",
+    "@types/chai": "^4.3.5",
     "@types/chai-as-promised": "^7.1.5",
     "@types/deasync": "^0.1.2",
     "@types/glob": "^8.1.0",
@@ -45,11 +45,12 @@
     "@types/node": "*",
     "chai": "^4.3.7",
     "chai-as-promised": "^7.1.1",
-    "glob": "^10.1.0",
+    "glob": "^10.2.3",
     "intermock": "^0.2.5",
     "mocha": "^10.2.0",
-    "nock": "^13.3.0",
-    "rollup": "^3.20.4",
+    "nock": "^13.3.1",
+    "rollup": "^3.21.6",
+    "rollup-plugin-dts": "^5.3.0",
     "ts-node": "^10.9.1",
     "typescript": "^5.0.4"
   }

--- a/rollup.config.js
+++ b/rollup.config.js
@@ -1,60 +1,82 @@
 import terser from '@rollup/plugin-terser';
 import typescript from '@rollup/plugin-typescript';
+import dts from "rollup-plugin-dts";
 import pkg from './package.json';
 
-export default {
-  input: 'src/TestRail.ts',
-  external: [],
-  output: [
-    {
-      file: pkg.browser,
-      format: 'es',
-      exports: 'default',
-      sourcemap: true,
-    },
-    {
-      file: pkg.jsdelivr,
-      format: 'umd',
-      exports: 'default',
-      sourcemap: true,
-      name: 'TestRail',
-    },
-    {
-      file: pkg.main,
+export default [
+  {
+    input: 'src/TestRail.ts',
+    external: ['stream'],
+    output: {
+      file: 'dist/TestRail.d.ts',
       format: 'cjs',
-      exports: 'default',
-      strict: false,
-      sourcemap: true,
-      banner: `'use strict';\n`
-        + `const http = require('http');\n`
-        + `const https = require('https');\n`
-        + `const nodeFetch = require('node-fetch');\n`
-        + `const FormData = require('form-data');\n\n`
-        + `const options = { keepAlive: true };\n`
-        + `const httpAgent = new http.Agent(options);\n`
-        + `const httpsAgent = new https.Agent(options);\n`
-        + `const agent = (url) => url.protocol === 'http:' ? httpAgent : httpsAgent;\n`
-        + `const fetch = (url, init) => nodeFetch(url, { agent, ...init });\n`,
+      footer: '\n// @ts-ignore\nexport = TestRail;',
     },
-    {
-      file: pkg.module,
-      format: 'es',
-      exports: 'default',
-      sourcemap: true,
-      banner: ``
-        + `import http from 'http';\n`
-        + `import https from 'https';\n`
-        + `import nodeFetch from 'node-fetch';\n`
-        + `import FormData from 'form-data';\n\n`
-        + `const options = { keepAlive: true };\n`
-        + `const httpAgent = new http.Agent(options);\n`
-        + `const httpsAgent = new https.Agent(options);\n`
-        + `const agent = (url) => url.protocol === 'http:' ? httpAgent : httpsAgent;\n`
-        + `const fetch = (url, init) => nodeFetch(url, { agent, ...init });\n`,
-    },
-  ],
-  plugins: [
-    typescript({ tsconfig: './tsconfig.json' }),
-    terser()
-  ],
-};
+    plugins: [
+      dts.default(),
+      {
+        generateBundle(_, bundle) {
+          for (const file of Object.values(bundle)) {
+            file.code = file.code.replace(/declare const ([\w$]+): typeof (\w+);/g, 'type $1 = $2;');
+          }
+        }
+      }
+    ],
+  },
+  {
+    input: 'src/TestRail.ts',
+    external: [],
+    output: [
+      {
+        file: pkg.browser,
+        format: 'es',
+        exports: 'default',
+        sourcemap: true,
+      },
+      {
+        file: pkg.jsdelivr,
+        format: 'umd',
+        exports: 'default',
+        sourcemap: true,
+        name: 'TestRail',
+      },
+      {
+        file: pkg.main,
+        format: 'cjs',
+        exports: 'default',
+        strict: false,
+        sourcemap: true,
+        banner: `'use strict';\n`
+          + `const http = require('http');\n`
+          + `const https = require('https');\n`
+          + `const nodeFetch = require('node-fetch');\n`
+          + `const FormData = require('form-data');\n\n`
+          + `const options = { keepAlive: true };\n`
+          + `const httpAgent = new http.Agent(options);\n`
+          + `const httpsAgent = new https.Agent(options);\n`
+          + `const agent = (url) => url.protocol === 'http:' ? httpAgent : httpsAgent;\n`
+          + `const fetch = (url, init) => nodeFetch(url, { agent, ...init });\n`,
+      },
+      {
+        file: pkg.module,
+        format: 'es',
+        exports: 'default',
+        sourcemap: true,
+        banner: ``
+          + `import http from 'http';\n`
+          + `import https from 'https';\n`
+          + `import nodeFetch from 'node-fetch';\n`
+          + `import FormData from 'form-data';\n\n`
+          + `const options = { keepAlive: true };\n`
+          + `const httpAgent = new http.Agent(options);\n`
+          + `const httpsAgent = new https.Agent(options);\n`
+          + `const agent = (url) => url.protocol === 'http:' ? httpAgent : httpsAgent;\n`
+          + `const fetch = (url, init) => nodeFetch(url, { agent, ...init });\n`,
+      },
+    ],
+    plugins: [
+      typescript({ tsconfig: './tsconfig.json' }),
+      terser()
+    ],
+  }
+];

--- a/rollup.config.js
+++ b/rollup.config.js
@@ -9,7 +9,7 @@ export default [
     external: ['stream'],
     output: {
       file: 'dist/TestRail.d.ts',
-      format: 'cjs',
+      format: 'es',
       footer: '\n// @ts-ignore\nexport = TestRail;',
     },
     plugins: [

--- a/src/types.d.ts
+++ b/src/types.d.ts
@@ -1,6 +1,0 @@
-export * from './TestRail';
-
-import TestRail from './TestRail';
-
-// @ts-ignore
-export = TestRail;

--- a/test/bdd.ts
+++ b/test/bdd.ts
@@ -1,5 +1,5 @@
 import { faker } from '@faker-js/faker';
-import type { AddAttachment, Case } from '../src/types';
+import type { AddAttachment, Case } from '..';
 import { OK, api, jsonFor, on } from './_helper';
 
 describe('BDD', () => {

